source/applications/advanced/color_balance.py

```diff
@@ -0,0 +1,257 @@
+        corrected_blue_balance / default_blue_balance,
+    )
+
+
+def _display_color_balance_example_results():
+    """Load and display example images before and after color balance.
+
+    Args None
+
+    Returns None
+
+    """
+    print(f"Displaying color balance calibration example results")
+    dir_path = Path(__file__).parent / "color_balance_files"
+    _display_rgb(
+        plt.imread(str(dir_path / "white_surface_before.png")),
+        "White surface before color balance",
+    )
+    _display_rgb(
+        plt.imread(str(dir_path / "white_surface_after.png")),
+        "White surface after color balance",
+    )
+    _display_rgb(
+        plt.imread(str(dir_path / "scene_before.png")), "Scene before color balance"
+    )
+    _display_rgb(
+        plt.imread(str(dir_path / "scene_after.png")), "Scene after color balance"
+    )
+
+
+def _set_settings_3d(camera, iris, exposure_time, brightness, gain):
+    """Set settings for 3D capture.
+
+    Args:
+        camera: Zivid camera
+        iris: Iris
+        exposure_time: Exposure time
+        brightness: Projector brightness
+        gain: Gain
+
+    Returns None
+
+    """
+    with camera.update_settings() as updater:
+        updater.settings.iris = iris
+        updater.settings.exposure_time = datetime.timedelta(microseconds=exposure_time)
+        updater.settings.brightness = brightness
+        updater.settings.gain = gain
+
+
+def _set_settings_2d(application, iris, exposure_time, brightness, gain):
+    """Set settings for 2D capture.
+
+    Args:
+        application: Zivid application
+        iris: Iris
+        exposure_time: Exposure time
+        brightness: Projector brightness
+        gain: Gain
+
+    Returns:
+        settings_2d: 2D capture settings
+
+    """
+    settings_2d = application.Settings2D()
+    settings_2d.iris = iris
+    settings_2d.exposure_time = datetime.timedelta(microseconds=exposure_time)
+    settings_2d.brightness = brightness
+    settings_2d.gain = gain
+
+    return settings_2d
+
+
+def _apply_color_balance(image, red_balance, blue_balance):
+    """Apply color balance to image.
+
+    Args:
+        image: Input image
+        red_balance: Red balance
+        blue_balance: Blue balance
+
+    Returns:
+        corrected_image: Image after color balance
+
+    """
+
+    corrected_image = image
+    image[:, :, 0] = corrected_image[:, :, 0] * np.float32(red_balance)
+    image[:, :, 2] = corrected_image[:, :, 2] * np.float32(blue_balance)
+
+    return corrected_image
+
+
+def _zivid_2d_frame_to_numpy_image(frame_2d):
+    """Convert Zivid 2D frame to numpy image.
+
+    Args:
+        frame_2d: Zivid 2D frame
+
+    Returns:
+        rgb: Numpy RGB image
+
+    """
+    rgb_zivid = frame_2d.image().to_array()
+    rgb = np.dstack([rgb_zivid["r"], rgb_zivid["g"], rgb_zivid["b"], rgb_zivid["a"]])
+
+    return rgb
+
+
+def _main():
+
+    _display_color_balance_example_results()
```

Comment:
Why is this part of the example?

Reply:
I wanted to demonstrate what the sample does. Perhaps this should be part of the Readme file for this sample.

Reply:
Yes, I would rather seperate these at least.

Reply:
I've removed it from the sample.